### PR TITLE
doc: adds faq page to site

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**

### CODE REVIEW

File: docs/docs/07-faq.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+### How do I write a Task Worker?
+
+Check out our [Task Worker Development Docs](./05-developer-guide/05-task-worker-development.md)!
+
+### How do I run a Workflow?
+
+You can run a workflow in two ways:
+* On the command line by [using `lhctl`](./05-developer-guide/03-lhctl.md#run-a-wfrun)
+* Programmatically by [using one of our SDK's](./05-developer-guide/09-grpc/10-running-workflows.md).
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:31]: * Programmatically by [using one of our SDK's](./05-developer-guide/09-grpc/10-running-workflows.md).  In the background the SDKs call a GRPC endpoint hosted on the LittleHorse server.

File: docs/docs/07-faq.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+The core concept of LittleHorse is a [Workflow](./04-concepts/01-workflows.md), which defines a series of tasks to execute and events to wait for. A `WfSpec` defines a workflow specification. A `WfRun` is a running instance of that `WfSpec`.
+
+### How do I create a `WfSpec`?
+
+Check out our [`WfSpec` development docs](./05-developer-guide/08-wfspec-development/08-wfspec-development.md)! In short, you can use our Java, Python, or Go SDK's to define the `WfSpec` logic and then register them to the LH Server.
+
+### What is a Task Worker?
+
+A [Task Worker](./04-concepts/03-tasks.md) is a program that opens a connection to a LittleHorse Cluster and listens to a task queue for a specific `TaskDef`. When a `WfRun` arrives at a point where it needs to execute that specific type of `TaskRun`, then the scheduled Task will be dispatched to the Task Worker. The Task Worker executes it and reports the result to the LH Server.
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:21]: what is a TaskDef?

File: docs/docs/07-faq.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+### How do I write a Task Worker?
+
+Check out our [Task Worker Development Docs](./05-developer-guide/05-task-worker-development.md)!
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:25]: A task worker is where tasks are executed.  
Check out our [Task Worker Development Docs](./05-developer-guide/05-task-worker-development.md)!

File: docs/docs/07-faq.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+### How do I run a Workflow?
+
+You can run a workflow in two ways:
+* On the command line by [using `lhctl`](./05-developer-guide/03-lhctl.md#run-a-wfrun)
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:30]: This is best suited for testing, development, or debugging.

File: docs/docs/07-faq.md (hunk 1 above)
Comment (mitchell-h) [reply docs/docs/07-faq.md:31]: Programmatic execution is ideal for incorporating into your dashboards, and operational tools.

File: docs/docs/07-faq.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+* Programmatically by [using one of our SDK's](./05-developer-guide/09-grpc/10-running-workflows.md).
+
+## Variables and Control Flow
+
+### How do I define variables in a Workflow?
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:35]: Move this below "What is a variable?"

File: docs/docs/07-faq.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+Check out the [Variables Section](./05-developer-guide/08-wfspec-development/01-basics.md#defining-variables) of our `WfSpec` docs.
+
+### Can I search for workflows by their variables?
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:39]: Move this below "### How do I pass information from one task to another?"

File: docs/docs/07-faq.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+[External Events](./04-concepts/04-external-events.md) in LittleHorse represent events that occur outside of the context of a single `WfRun`. For example, you might use an `ExternalEvent` to notify a `WfRun` when someone signs a DocuSign document, or when a text message is replied to.
+
+An `ExternalEvent` is an instance of an `ExternalEventDef`.
+
+### How do I make a `WfRun` wait for an event?
+
+Check out the [External Events](./05-developer-guide/08-wfspec-development/04-external-events.md) section of our `WfSpec` docs. What you want is an `ExternalEventNode`, which blocks until an `ExternalEvent` arrives.
+
+### How do I interrupt a `WfRun`?
+
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:68]: Why would you interrupt a `wfRun`?

File: docs/docs/07-faq.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+[User Tasks](./04-concepts/05-user-tasks.md) allow a workflow to wait for input from a human.
+
+### How do I add a user task to my workflow?
+
+Check out the [User Tasks](./05-developer-guide/08-wfspec-development/08-user-tasks.md) page of our `WfSpec` docs.
+
+### Do User Tasks support users and user groups?
+
+Yes, user tasks can be assigned to either a user_id or a user_group. Note that user and group identities are not managed by the LH Server.
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:121]: if they're not managed by LH server, where are they managed?

File: docs/docs/07-faq.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+## User Tasks
+
+### What are User Tasks?
+
+[User Tasks](./04-concepts/05-user-tasks.md) allow a workflow to wait for input from a human.
+
+### How do I add a user task to my workflow?
+
+Check out the [User Tasks](./05-developer-guide/08-wfspec-development/08-user-tasks.md) page of our `WfSpec` docs.
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:117]: Why would I add a user task to my workflow?

File: docs/docs/07-faq.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+### How do I throw an `EXCEPTION`?
+
+You can [throw an `EXCEPTION` from a Task Worker](./05-developer-guide/05-task-worker-development.md#throwing-workflow-exceptions). You can also make a `ThreadRun` fail with the `WorkflowThread.fail()` method.
+
+## Reliability
+
+### Is LittleHorse Fault-Tolerant?
+
+Yes, it was built into the [core DNA of our system](./02-architecture-and-guarantees.md).
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:149]: Every component of the LittleHorse tech stack is highly available and fault tolerant.  At the workflow layer LittleHorse provides the primitives to handle faults in a way to suit the business needs.

File: docs/docs/07-faq.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+### Does LittleHorse scale?
+
+Yes.
```
Comment (mitchell-h) [inline docs/docs/07-faq.md:153]: Yes, in our internal benchmarks a moderately sized, single instance, of LittleHorse server can execute 30,000 tasks per second in a moderately complex workflow.

**Review by mitchell-h — APPROVED** [review]

**Review by eduwercamacaro — APPROVED** [review]

**Pull request merged**